src/analysis/potential_analysis.jl

```diff
@@ -0,0 +1,209 @@
+                            Z = [z0, z1, z2], pbc = false, #
+                            cell = [ 1.0 + maximum([r1, r2]) 0.0 0.0;
+                            0.0 (1 + r2) 0.0;
+                            0.0 0.0 1])
+
+"""
+`function atom_energy(IP, z0)` : energy of an isolated atom
+"""                            
+function atom_energy(IP, z0)
+   at = Atoms(X = [SVector(0.0,0.0,0.0),], Z = [z0, ], 
+              pbc = false, cell = [1.0 0 0; 0 1.0 0; 0 0 1.0])
+   return energy(IP, at)                  
+end
+
+"""
+`function dimer_energy(pot, r, z1, z0)` : energy of a dimer 
+with separation `r` and atomic numbers `z1` and `z0` using the potential `pot`; 
+subtracting the 1-body contributions. 
+"""
+function dimer_energy(IP, r, z1, z0)
+   at = at_dimer(r, z1, z0)
+   return energy(IP, at) - atom_energy(IP, z0) - atom_energy(IP, z1)
+end
+
+"""
+`function trimer_energy(IP, r1, r2, θ, z0, z1, z2)` : computes the energy of a
+trimer, subtracting the 2-body and 1-body contributions.
+"""
+function trimer_energy(IP, r1, r2, θ, z0, z1, z2)
+   at = at_trimer(r1, r2, θ, z0, z1, z2)
+   dr1r2 = sqrt(r1 ^ 2 + r2 ^ 2 - 2 * r1 * r2 * cos(θ))
+   return ( energy(IP, at) 
+            - dimer_energy(IP, r1, z1, z0) 
+            - dimer_energy(IP, r2, z2, z0) 
+            - dimer_energy(IP, dr1r2, z1, z2)
+            - atom_energy(IP, z0) 
+            - atom_energy(IP, z1) 
+            - atom_energy(IP, z2) )
+end
+
+# TODO: this needs a PR to JuLIP 
+_cutoff(potential) = cutoff(potential)
+_cutoff(potential::JuLIP.MLIPs.SumIP) = maximum(_cutoff.(potential.components))
+_cutoff(potential::JuLIP.OneBody) = 0.1   # not sure how 0.0 will behave
+
+
+"""
+Generate a dictionary of dimer curves for a given potential. 
+"""
+function dimers(potential, elements; 
+                rr = range(1e-3, _cutoff(potential), length=200), 
+                minE = -1e5, maxE = 1e5, )
+   zz = AtomicNumber.(elements)
+   dimers = Dict() 
+   for i = 1:length(zz), j = 1:i
+      z1 = zz[i]
+      z0 = zz[j]
+      v01 = dimer_energy.(Ref(potential), rr, z1, z0)
+      v01 = max.(min.(v01, maxE), minE)
+      dimers[(z1, z0)] = (rr, v01)
+   end
+   copy_zz_sym!(dimers)
+   return dimers
+end
+
+"""
+Generate a dictionary of trimer curves for a given potential.
+"""
+function trimers(potential, elements, r1, r2; 
+                θ = range(-pi, pi, length = 200), 
+                minE = -1e5, maxE = 1e5, )
+   zz = AtomicNumber.(elements)
+   trimers = Dict() 
+   for i = 1:length(zz), j = 1:i, k = 1:j
+      z0 = zz[i]
+      z1 = zz[j]
+      z2 = zz[k]
+      v01 = trimer_energy.(Ref(potential), r1, r2, θ, z0, z1, z2)
+      v01 = max.(min.(v01, maxE), minE)
+      trimers[(z0, z1, z2)] = (θ, v01)
+   end
+   copy_zz_sym!(trimers)
+   return trimers
+end
+
+# function trimers(potential, elements; r1 = 0.5:3:_cutoff(potential), r2 = 0.5:3:_cutoff(potential),
```

Comment:
Should we remove this before the merge?

Reply:
I wasn't sure which variant of the trimer function to use ... but this might be easy enough to recover with a few minutes' work. so yes I'll get rid of it too.